### DISCUSSION TITLE
Check connection function in Sync

### DISCUSSION THREAD

**Discussion by ingdesarr1** [FreeOpcUa/opcua-asyncio#1579] [Q&A]
Good afternoon. I want to use the function check_connection in the asyncua.sync to check if my client is connected, but reviewing the documentation I can't find any information and it seems that the function only returns None. Is there a way to use this function without having to change everything to async? since I will have to change my entire application and it is a lot of work. Thanks

**Comment by schroeder-** [comment]
Does `check_connection` throws an exception, it does not return a bool.

**Reply by ingdesarr1** [reply]
With sync, the function return None, and if you see the library, the function is void

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/146734555/950ad3ca-487c-4e64-a5ab-f01e287409d7)

**Reply by schroeder-** [reply]
No it is not. The `@syncmethod` decorator will call the async function. So it will work. 
Just do this:
```
try:
      client.check_connection()
except Exception as e:
      print(f"error in connection: {e}")
```

**Reply by ingdesarr1** [reply]
Not work. I try this:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/146734555/bbe3534a-d7ea-448c-944f-cc805afe6745)
I'm not connected to any opc-ua, and print this:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/146734555/ac022ff7-1de7-4aa9-a96e-7a55438cfb57)

**Reply by schroeder-** [reply]
`check_connection` only works if you have already connected to a server.

**Reply by ingdesarr1** [reply]
Thanks

**Reply by ingdesarr1** [reply]
Today, I try to use the function the server connected and the function return None. Is any other way to check the connection?

**Reply by schroeder-** [reply]
This function is to check if the connection is alive and not broken. If the connection is broken, the underlying exception is thrown.

If you want to know if you already connected, just add a bool and set it to true if you connected to the server.
